File: solution/array/274-h-index/solution.java
import java.util.Arrays;

class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);

        int n = citations.length;
        int h = 0;

        for (int i = n - 1; i >= 0; i--) {
            int papersWithAtLeastCurrentCitation = n - i;

            if (citations[i] >= papersWithAtLeastCurrentCitation) {
                h = papersWithAtLeastCurrentCitation;
            } else {
                break;
            }
        }

        return h;
    }
}

File: solution/array/274-h-index/solution.go
package main

import "sort"

func hIndex(citations []int) int {
	sort.Sort(sort.Reverse(sort.IntSlice(citations)))

	h := 0

	for i, citation := range citations {
		if citation >= i+1 {
			h = i + 1
		} else {
			break
		}
	}

	return h
}